src/Microsoft.ComponentDetection.Common/PathUtilityService.cs

```diff
@@ -117,176 +53,14 @@ public bool MatchesPattern(string searchPattern, string fileName)
-        }
-
-        var pointer = IntPtr.Zero;
-        try
-        {
-            pointer = RealPathLinux(path, IntPtr.Zero);
-
-            if (pointer != IntPtr.Zero)
-            {
-                var toReturn = Marshal.PtrToStringAnsi(pointer);
-                return toReturn;
-            }
-            else
-            {
-                return path;
-            }
-        }
-        catch (Exception ex)
-        {
-            this.logger.LogError(ex, "Failed to resolve path {Path}", path);
-            return path;
-        }
-        finally
-        {
-            if (pointer != IntPtr.Zero)
-            {
-                FreeMemoryLinux(pointer);
-            }
-        }
-    }
-
-    [DllImport("kernel32.dll", EntryPoint = "CreateFileW", CharSet = CharSet.Unicode, SetLastError = true)]
-    [DefaultDllImportSearchPaths(DllImportSearchPath.System32)]
-    private static extern SafeFileHandle CreateFile(
-        [In] string lpFileName,
-        [In] uint dwDesiredAccess,
-        [In] uint dwShareMode,
-        [In] IntPtr lpSecurityAttributes,
-        [In] uint dwCreationDisposition,
-        [In] uint dwFlagsAndAttributes,
-        [In] IntPtr hTemplateFile);
-
-    [DllImport("kernel32.dll", EntryPoint = "GetFinalPathNameByHandleW", CharSet = CharSet.Unicode, SetLastError = true)]
-    [DefaultDllImportSearchPaths(DllImportSearchPath.System32)]
-    private static extern int GetFinalPathNameByHandle([In] IntPtr hFile, [Out] char[] lpszFilePath, [In] int cchFilePath, [In] int dwFlags);
-
-    private bool CheckIfRunningOnWindowsContainer()
-    {
-        if (IsLinux)
-        {
-            return false;
-        }
-
-        // This isn't the best way to do this in C#, but netstandard doesn't seem to support the service api calls
-        // that we need to do this without shelling out
-        var process = new Process()
-        {
-            StartInfo = new ProcessStartInfo
-            {
-                FileName = "cmd.exe",
-                Arguments = "/c NET START",
-                CreateNoWindow = true,
-                RedirectStandardOutput = true,
-                RedirectStandardError = false,
-                UseShellExecute = false,
-            },
-        };
-
-        var sb = new StringBuilder();
-        process.Start();
-
-        while (!process.HasExited)
-        {
-            sb.Append(process.StandardOutput.ReadToEnd());
-        }
-
-        process.WaitForExit();
-        sb.Append(process.StandardOutput.ReadToEnd());
-
-        if (sb.ToString().Contains("Container Execution Agent"))
-        {
-            this.logger.LogWarning("Detected execution in a Windows container. Currently windows containers < 1809 do not support symlinks well, so disabling symlink resolution/dedupe behavior");
-            return true;
-        }
-        else
-        {
-            return false;
-        }
-    }
+    public string ResolvePhysicalPath(string path) => Path.GetFullPath(path);
```

Comment:
Does `GetFullPath` follow symlinks?

Reply:
Yes: 
<img width="574" alt="image" src="https://user-images.githubusercontent.com/26028834/223215868-b9cbce98-5d79-4ed3-9b80-29f883adc722.png">

Reply:
It does not. I've updated it to properly create either a `FileInfo` or `DirectoryInfo` class, then we can use the [`LinkTarget`](https://learn.microsoft.com/en-us/dotnet/api/system.io.filesysteminfo.linktarget?view=net-7.0) property.